### ISSUE TITLE
Recommended way for setting up multiple apps on the same server as wallabag

### ISSUE THREAD

**Issue by joelostblom** [wallabag/wallabag#3324]
Posted this as a question on the gitter first, but trying here instead since there was no reply.

### Issue details

I have installed wallabag on my server using the "shared hosting" approach: `wget https://wllbg.org/latest-v2-package && tar xvf latest-v2-package` (I could not get either git setup or the docker image to work). I also have miniflux installed for RSS reading and I would like to access both these apps. 

Miniflux has [straightforward instructions](https://miniflux.net/documentation/installation), I basically clone the GitHub repo and then navigate to `localhost/miniflux` to login and access my feeds. With wallabag, I only get a directory listing when navigating to `localhost/wallabag` after cloning and running `make install`. In order to get the wallabag login interface, I can type `make run` which launches the PHP server, but since this is not recommend for production use, I tried setting up the apache virtual host. I ran into permission troubles in the beginning, but I got it working eventually. A new issue arose and I could not access the root server location `localhost`, since it redirected me to the wallabag login page. To fix this I changed the the `DocumentRoot` field to `/var/www/html/` in `/etc/apache2/sites-available/000-default.conf`.

It is great that I can now access the wallabag login page at `localhost/wallabag/web` and other apps via `localhost/<app_name>`. But since I am completely new to self-hosting, and ran into multiple errors when setting up, I want to ask if my current solution is a good way of accessing several apps alongside wallbag, or if there is some specific reason that wallabag should redirect the base server url to its login page, and there another approach to achieve what I want.

I have all my app folders in `/var/www/html/`. My `/etc/apache2/sites-available/000-default.conf` looks like this

```
<VirtualHost *:80>
    ServerName domain.tld
    ServerAlias www.domain.tld

    DocumentRoot /var/www/html
    <Directory /var/www/html/wallabag/web>
        Require all granted
        <IfModule mod_rewrite.c>
            Options -MultiViews
            RewriteEngine On
            RewriteCond %{REQUEST_FILENAME} !-f
            RewriteRule ^(.*)$ app.php [QSA,L]
        </IfModule>
    </Directory>

    # uncomment the following lines if you install assets as symlinks
    # or run into problems when compiling LESS/Sass/CoffeScript assets
    # <Directory /var/www/html/wallabag>
    #     Options FollowSymlinks
    # </Directory>

    # optionally disable the RewriteEngine for the asset directories
    # which will allow apache to simply reply with a 404 when files are
    # not found instead of passing the request into the full symfony stack
    <Directory /var/www/html/wallabag/web/bundles>
        <IfModule mod_rewrite.c>
            RewriteEngine Off
        </IfModule>
    </Directory>
    ErrorLog /var/log/apache2/wallabag_error.log
    CustomLog /var/log/apache2/wallabag_access.log combined
</VirtualHost>
```

### Environment

* wallabag version (or git revision) that exhibits the issue: **2.2.3**
* How did you install wallabag? Via `git clone` or by downloading the package? **Downloading the package**
* Last wallabag version that did not exhibit the issue (if applicable): **NA**
* php version: **5.6.30**
* OS: **Antergos Linux (Arch derivative)**
* type of hosting (shared or dedicated):
* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL): **MariaDB**

### Steps to reproduce/test case

See details

**Comment by keyslapper** [comment]
I don't suppose anyone has found a solution for this?  I've used Wallabag for years, and have always used it alongside several other applications.

**Comment by techexo** [comment]
Hi. This issue came a lot on GitHub, and it is specified in the doc: wallabag currently doesn't officially support installation into a subfolder (e.g. `example.com/wallabag`), only at the root of the domain (`example.com`) or subdomain (`wallabag.example.com`).

In your case, I would suggest using sub-domains (search for Apache Virtualhosts configurations on your favorite search engine) if you can. I cannot help you more on this because I don't use Apache, but there is lots of tutorials about this, and I can assure you that you'll be able to have plenty of apps running in parallel 🙂 (I have maybe 6 or 7 services running on my self-hosted server, each in a subdomain/virtual host and it's perfect).

**Comment by keyslapper** [comment]
Thanks for the feedback, but this complicates keeping my services on https.  I'm using a letsencrypt certificate for my www subdomain, and Wallabag is my 6th service on this server.

I guess I'm going to have to muddle through until I can figure out whether I can get a wildcard cert and how to configure it for my services.

Cheers!
Lou

**Comment by techexo** [comment]
Hi @keyslapper . The bot for let's encrypt certificate (certbot on Linux for most distributions) is able to read Virtualhosts configurations and make one certificate valid for all configured sub-domains. So you won't need to have multiple certs to manage, just the one, an when you add a service you just update your cert with certbot!

**Comment by JialuZhang** [comment]
@joelostblom 

@GoogleCodeExporter

Thanks for the sharing! However, under the directory of 
```
<Directory /var/www/html/wallabag/web/bundles>
        <IfModule mod_rewrite.c>
```
your posted configuration, the line: "RewriteEngine Off" is a misconfiguration, and adding it to your system will not change any system behavior.
Why "RewriteEngine Off" is allowed by Apache is that, if you include multiple "RewriteRule" parameters in your configuration, then instead of commenting them all, you can explicitly using “RewriteEngine Off” to disable all "RewriteRule".

More importantly, the default value of “RewriteEngine" is already an "off", so adding “RewriteEngine Off" is quite unnecessary and it may cause confusion to users.

Since herein there is no "RewriteRule", deleting “RewriteEngine Off” would be ideal.

Related Apache source code snippet:
```
run_rewritemap_programs(server_rec *s , apr_pool_t *p){
if (conf->state == ENGINE_DISABLED) { // usage of "RewriteEngine"
  return APR_SUCCESS; // early return
rewritemap_program(...); // usage of "RewriteRule" 
}
```
